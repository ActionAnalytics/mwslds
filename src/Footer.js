import React from 'react'

import backToTop from 'Gov-2.0-Bootstrap-Skeleton/dist/images/back-to-top.png'

import './bcgov_bootstrap'

const genericLinks = [
  {
    name: 'Home',
    href: '#',
  },
  {
    name: 'About',
    href: '#',
  },
  {
    name: 'Disclaimer',
    href: 'https://www2.gov.bc.ca/gov/content/home/disclaimer',
  },
  {
    name: 'Privacy',
    href: 'https://www2.gov.bc.ca/gov/content/home/privacy',
  },
  {
    name: 'Accessibility',
    href: 'https://www2.gov.bc.ca/gov/content/home/accessibility',
  },
  {
    name: 'Copyright',
    href: 'https://www2.gov.bc.ca/gov/content/home/copyright',
  },
  {
    name: 'Contact Us',
    href: 'https://www2.gov.bc.ca/gov/content/home/contact-us',
  },
]

export default function Footer(props) {
  const { items } = props

  const allItems = genericLinks.concat(items || [])

  const itemElements = allItems.map(link => (
    <li key={link.name}><a className="nav-item nav-link" href={link.href}>{link.name}</a></li>
  ))

  return (
    <div id="footer">
      <img className="back-to-top footer-overlap" src={backToTop} alt="Back to top" />
      <div id="footerWrapper">
        <div id="footerAdminSection">
          <div id="footerAdminLinksContainer" className="container">
            <div id="footerAdminLinks" className="row">
              <ul className="inline">
                {itemElements}
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}
