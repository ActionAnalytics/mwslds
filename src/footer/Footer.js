import React from 'react'
import PropTypes from 'prop-types'

// TODO: import images like this from the bcgov_bootstrap wrapper
import backToTop from 'Gov-2.0-Bootstrap-Skeleton/dist/images/back-to-top.png'

import '../style'

const propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    href: PropTypes.string,
  })),
}

const defaultProps = {
  items: [],
}

const requiredLinks = [
  {
    name: 'Home',
    href: '#',
  },
  {
    // TODO: handle this better... even though the bcgov one is broken they say its required
    name: 'About',
    href: '#',
  },
  {
    name: 'Disclaimer',
    href: 'https://www2.gov.bc.ca/gov/content/home/disclaimer',
  },
  {
    name: 'Privacy',
    href: 'https://www2.gov.bc.ca/gov/content/home/privacy',
  },
  {
    name: 'Accessibility',
    href: 'https://www2.gov.bc.ca/gov/content/home/accessibility',
  },
  {
    name: 'Copyright',
    href: 'https://www2.gov.bc.ca/gov/content/home/copyright',
  },
  {
    name: 'Contact Us',
    href: 'https://www2.gov.bc.ca/gov/content/home/contact-us',
  },
]

function Footer(props) {
  const { items } = props

  const allItems = requiredLinks.concat(items || [])

  const itemElements = allItems.map(link => (
    <li key={link.name}><a className="nav-item nav-link" href={link.href}>{link.name}</a></li>
  ))

  return (
    <div id="footer">
      <img className="back-to-top footer-overlap" src={backToTop} alt="Back to top" />
      <div id="footerWrapper">
        <div id="footerAdminSection">
          <div id="footerAdminLinksContainer" className="container">
            <div id="footerAdminLinks" className="row">
              <ul className="inline">
                {itemElements}
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

Footer.propTypes = propTypes
Footer.defaultProps = defaultProps

export default Footer
