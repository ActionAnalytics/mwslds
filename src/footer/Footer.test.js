import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import { createSerializer } from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import Footer from './Footer'

configure({ adapter: new Adapter() })
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))

describe('Footer', () => {
  it('renders', () => {
    const header = shallow(<Footer />)
    expect(header).toMatchSnapshot()
  })

  it('renders additional items correctly', () => {
    const items = [
      { name: 'far away', href: '/away' },
      { name: 'close to home', href: '/close' },
    ]
    const header = mount(<Footer items={items} />)
    expect(header).toMatchSnapshot()
  })
})
