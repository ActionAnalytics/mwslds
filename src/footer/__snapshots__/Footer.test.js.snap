// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<div
  id="footer"
>
  <img
    alt="Back to top"
    className="back-to-top footer-overlap"
    src="back-to-top.png"
  />
  <div
    id="footerWrapper"
  >
    <div
      id="footerAdminSection"
    >
      <div
        className="container"
        id="footerAdminLinksContainer"
      >
        <div
          className="row"
          id="footerAdminLinks"
        >
          <ul
            className="inline"
          >
            <li
              key="Home"
            >
              <a
                className="nav-item nav-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              key="About"
            >
              <a
                className="nav-item nav-link"
                href="#"
              >
                About
              </a>
            </li>
            <li
              key="Disclaimer"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              >
                Disclaimer
              </a>
            </li>
            <li
              key="Privacy"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/privacy"
              >
                Privacy
              </a>
            </li>
            <li
              key="Accessibility"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              key="Copyright"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/copyright"
              >
                Copyright
              </a>
            </li>
            <li
              key="Contact Us"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              >
                Contact Us
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer renders additional items correctly 1`] = `
<div
  id="footer"
>
  <img
    alt="Back to top"
    className="back-to-top footer-overlap"
    src="back-to-top.png"
  />
  <div
    id="footerWrapper"
  >
    <div
      id="footerAdminSection"
    >
      <div
        className="container"
        id="footerAdminLinksContainer"
      >
        <div
          className="row"
          id="footerAdminLinks"
        >
          <ul
            className="inline"
          >
            <li
              key="Home"
            >
              <a
                className="nav-item nav-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              key="About"
            >
              <a
                className="nav-item nav-link"
                href="#"
              >
                About
              </a>
            </li>
            <li
              key="Disclaimer"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              >
                Disclaimer
              </a>
            </li>
            <li
              key="Privacy"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/privacy"
              >
                Privacy
              </a>
            </li>
            <li
              key="Accessibility"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              key="Copyright"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/copyright"
              >
                Copyright
              </a>
            </li>
            <li
              key="Contact Us"
            >
              <a
                className="nav-item nav-link"
                href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              >
                Contact Us
              </a>
            </li>
            <li
              key="far away"
            >
              <a
                className="nav-item nav-link"
                href="/away"
              >
                far away
              </a>
            </li>
            <li
              key="close to home"
            >
              <a
                className="nav-item nav-link"
                href="/close"
              >
                close to home
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
